blockchain/blockchain.go

```diff
@@ -1632,31 +1638,47 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 		if err != nil {
 			return i, events, coalescedLogs, err
 		}
+		// If we have a followup block, run that against the current state to pre-cache
+		// transactions and probabilistically some of the account/storage trie nodes.
+		bc.followupInterrupt = 0
+
+		if !bc.cacheConfig.TrieNodeCacheConfig.NoPrefetch {
+			if i == 0 {
+				for k := i + 1; k < len(chain); k++ {
+					bc.prefetchCh <- chain[k]
+				}
+				logger.Info("sent blocks to prefetch channel", "numBlocks", len(chain)-1)
+			}
+		}
 
 		// Process block using the parent state as reference point.
 		receipts, logs, usedGas, internalTxTraces, procStats, err := bc.processor.Process(block, stateDB, bc.vmConfig)
 		if err != nil {
 			bc.reportBlock(block, receipts, err)
+			atomic.StoreUint64(&bc.followupInterrupt, 1)
```

Comment:
@ehnuje 
This code stop the prefetching future block after every one block.
I think this makes very frequent context switch.
So I suggest to start prefetching all blocks(chain) before `for i, block := range chain {` like `senderCacher()`.
Also we can stop the fetch goroutine after this loop.

I would like you to test this fetch logic also.

Reply:
As this PR showed worse performance than only fetching next block, I'll just update this PR and compare its performance to the original prefetcher.

Reply:
Please check 21fbc537c66f705af24f5c2b615c378c3c6bebb8